platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/ValueNodeMap.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Tracks the values of inputs and selects on the server and client side.
+ */
+class ValueNodeMap {
+	private readonly states = new Map<HTMLElement, ValueNodeState>();
+
+	/**
+	 * Adds the given value node to this map.
+	 */
+	public addNode(node: HTMLElement) {
+		this.states.set(node, new ValueNodeState(node));
+	}
+
+	/**
+	 * Approves all pending values.
+	 */
+	public approveNodeValues() {
+		this.states.forEach((state, node) => state.approveValue());
+	}
+
+	/**
+	 * Transfers all changed values of value nodes to the specified data structure.
+	 */
+	public copyNodeValues(data: any) {
+		this.states.forEach((state, node) => state.updatePendingValue(data));
+	}
+
+	public isValueChanged(node: HTMLElement) {
+		return this.getState(node).isValueChanged();
+	}
+
+	public setValue(node: HTMLInputElement, value: string) {
+		node.value = value;
+		this.getState(node).assumeValue(value);
+
+		// FIXME this ad-hoc call looks like a hack
```

Comment:
Do we already have a story to re-implement ACs? If so, we could reference it here.

Reply:
added reference to PLAT-853